### ISSUE TITLE
Make data attributes more semantic

### ISSUE THREAD

**Issue by patcon** [pol-is/issues#78]
Reticketed from this: https://github.com/pol-is/issues/issues/77

Heyo! Terseness is sometimes nice (eg. when typing flags in the command line, when speed is important), but it's sometimes obfuscates meaning.

I would suggest that `data-` attributes don't need to be terse as much as they should to be legible :)

Could we provide expanded data attributes, and direct folks via docs to use those instead?

I would suggest these todo's for closure (if you feel the proposal is a step forward):
- [ ] alias expanded attributes to same code path as current attr
- [ ] update docs to show expanded attributes
- [ ] refer to condensed attributes in docs, but clearly reference as deprecated
- [ ] (optional) add all attributes to doc, but note as deprecated those that you have decided to de-emphasize. (this allows interested folks to understand what old implementations are doing, but properly informs them that feature is going away, and they shouldn't build on it. If this is a concern or someone has a valid use-case, they will be able to more clearly understand the landscape and open up discussion with you about your future plan :) )

cc: @virgile-dev curious your thoughts on this ticket :)

**Comment by colinmegill** [comment]
Interesting, I would generally agree here. This would be something like
userCanVote="true" then. I would imagine we keep both syntaxes.
@mbjorkegren could elaborate on the constraints and motivations there.

**Comment by patcon** [comment]
Cool! Thanks for hearing me out on this :)

These two resources make me feel that `data-user-can-vote` is the most appropriate and expected convention. (afaict, html5 spec doesn't do camelcase for attributes.)

- https://stackoverflow.com/a/20985285/504018
- https://stackoverflow.com/a/22753630/504018

tl;dr: In HTML5 DOM API and jQuery, `data-userCanVote` becomes `usercanvote`, while `data-user-can-vote` becomes `userCanVote`.

**Comment by mbjorkegren** [comment]
These params get used in the URL of the page, and URLs do have a length limit (something like 2000 characters). Since we don't know how many params will be added in the future, I'm not too keen on lengthening these.
We have 31 parameters already, including one param that is itself a URL (data-x_profile_image_url). xid should be a long string.  
I want to be sure we leave space for additional long params.

**Comment by mbjorkegren** [comment]
Just thought of a solution to that, we could accept the long form for the data- attributes, but use the short form everywhere else.

**Comment by virgile-dev** [comment]
Hey sorry for reacting so late,
Seems like @mbjorkegren found a good compromise on this one :) accept long but keep short seems perfect to me. 
Can't help but notice, **31 attributes !** is there a full list somewhere ?
This ressource from @patcon https://github.com/pol-is/polisClientParticipation/blob/master/js/util/utils.js#L301-L369 was helpful to understand some of the attributes I couldn't understand.

**Comment by virgile-dev** [comment]
Hey @mbjorkegren ,

Looking at the code I found this attribute `bg_white`. 
Inside [api/embed.js](https://github.com/pol-is/polisClientParticipation/blob/4a32a6cf2eed7ca403b9ecbf9b4c3fd9e3c4a5df/api/embed.js) 
```
bg_white: d.getAttribute("data-bg_white"),
```
I tried to use it to figure out what it does but couldn't get anywhere. What kind of value is it supposed to be given ``true`` or ``false`` and when false what happens ?

Thanks a lot !

**Comment by mbjorkegren** [comment]
That one is deprecated. We added it back when we had a gray background to make it possible to use a white background.

**Comment by patcon** [comment]
Thought of this issue and "semantic"ness when reading this issue: https://github.com/pol-is/polis/issues/173

Would a little prefix or suffix help for parameters that work this way? maybe like `init`?

**Comment by patcon** [comment]
> @colinmegill: Interesting, I would generally agree here. This would be something like
userCanVote="true" then.

Seems we can't use camelCase in data attributes

> The name of a custom data attribute in HTML begins with data-. It must contain only letters, numbers and the following characters: dash (-), dot (.), colon (:), underscore (_) -- but **NOT any ASCII capital letters (A to Z)**. -- [MDN Web Docs: HTMLOrForeignElement.dataset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset)
